packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/zoomlevel/CameraZoom.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera.features.zoomlevel;
+
+import android.graphics.Rect;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.math.MathUtils;
+
+public final class CameraZoom {
+  public static final float DEFAULT_ZOOM_FACTOR = 1.0f;
+
+  @NonNull private final Rect cropRegion = new Rect();
+  @Nullable private final Rect sensorSize;
+
+  public final float maxZoom;
+  public final boolean hasSupport;
+
+  public static CameraZoom create(@Nullable final Rect sensorArraySize, final Float maxZoom) {
+    return new CameraZoom(sensorArraySize, maxZoom);
+  }
+
+  private CameraZoom(@Nullable final Rect sensorArraySize, final Float maxZoom) {
+    this.sensorSize = sensorArraySize;
+
+    if (this.sensorSize == null) {
+      this.maxZoom = DEFAULT_ZOOM_FACTOR;
+      this.hasSupport = false;
+      return;
+    }
+
+    this.maxZoom =
+        ((maxZoom == null) || (maxZoom < DEFAULT_ZOOM_FACTOR)) ? DEFAULT_ZOOM_FACTOR : maxZoom;
```

Comment:
Why does the default change the max, rather than the max changing the default?

Reply:
This statement is part of the constructor where one can pass in a parameter specifying the maximum zoom level. However if the specified maximum zoom level if not specified or below the minimum zoom level (which is named incorrectly and something I will fix), the maximum zoom level will be set equal to the minimum zoom level.